onedal/linear_model/incremental_linear_model.py

```diff
@@ -77,22 +77,18 @@ def partial_fit(self, X, y, queue=None):
         if not hasattr(self, "_policy"):
             self._policy = self._get_policy(queue, X)
 
+        X, y = _convert_to_supported(self._policy, X, y)
+
         if not hasattr(self, "_dtype"):
             self._dtype = get_dtype(X)
             self._params = self._get_onedal_params(self._dtype)
 
-        if self._dtype not in [np.float32, np.float64]:
-            self._dtype = np.float64
-
-        X = X.astype(self._dtype, copy=self.copy_X)
-        y = y.astype(dtype=self._dtype)
+        y = np.asarray(y).astype(dtype=self._dtype)
         self._y_ndim_1 = y.ndim == 1
 
-        X, y = _check_X_y(X, y, force_all_finite=False, accept_2d_y=True)
```

Comment:
force_all_finite occurs here for partial fit (https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/linear_model/incremental_linear.py#L157) Ideally we would disable the finite check there to allow for a single finite check.

Reply:
Recommended change, add force_all_finite=False to https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/linear_model/incremental_linear.py#L157, https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/linear_model/incremental_linear.py#L166 and https://github.com/intel/scikit-learn-intelex/blob/main/sklearnex/linear_model/incremental_linear.py#L171